test/lib/CMakeLists.txt

```diff
@@ -55,12 +55,15 @@ foreach(script ${test_script_files})
     add_test(
         NAME ${target}
         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-        COMMAND ${MAYA_PY_EXECUTABLE} -c "from unittest import main; \
+        COMMAND ${MAYA_PY_EXECUTABLE} -c "import sys; \
+                                          from unittest import main; \
                                           import maya.standalone; \
                                           maya.standalone.initialize(name='python'); \
                                           import ${target}; \
-                                          main(module=${target}); \
-                                          maya.standalone.uninitialize()"
+                                          testProg = main(module=${target}, exit=False); \
+                                          maya.standalone.uninitialize(); \
+                                          sys.exit(not testProg.result.wasSuccessful());
+                                          "
     )
     set_property(TEST ${target} APPEND PROPERTY ENVIRONMENT
         "PATH=${path}"
```

Comment:
@mattyjams, looks good to me.  Thank you.

On another note, we now have a new utility function "separate_argument_list" that was recently added to utils.cmake and deals with those crazy ";" ":" stuff on different platforms.
  
Please replace line 42 to 51 with these two call:

```
separate_argument_list(path)
separate_argument_list(pythonPath)
```

Reply:
Ahh, good call! I opened #218 for that change.

Reply:
Thanks.